### ISSUE TITLE
regression in vmagent 1.95.0: -dryRun no longer fails for invalid configuration under scrape_config_files

### ISSUE THREAD

**Issue by lauri-paypay** [VictoriaMetrics/VictoriaMetrics#5508]
### Describe the bug

Since https://github.com/VictoriaMetrics/VictoriaMetrics/pull/5153, `vmagent -dryRun` exits with status 0 even if the provided configuration is invalid. We use this in CI to check validity of our configuration before deploying.

### To Reproduce

```
echo "asdf" > foo.yaml
echo "scrape_config_files: [foo.yaml]" > promscrape.yaml
docker run --rm -v $(pwd):/tmp victoriametrics/vmagent:v1.96.0-enterprise -eula -dryRun -promscrape.config=/tmp/promscrape.yaml
```
```
2023-12-21T08:28:44.248Z	info	VictoriaMetrics/lib/logger/flag.go:12	build version: vmagent-20231212-225303-tags-v1.96.0-enterprise-0-ga771bc255
2023-12-21T08:28:44.248Z	info	VictoriaMetrics/lib/logger/flag.go:13	command-line flags
2023-12-21T08:28:44.248Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -dryRun="true"
2023-12-21T08:28:44.248Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -eula="true"
2023-12-21T08:28:44.248Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -promscrape.config="/tmp/promscrape.yaml"
2023-12-21T08:28:44.248Z	info	VictoriaMetrics/lib/license/copyrights.go:20	Copyright 2018-2023 VictoriaMetrics, Inc - All Rights Reserved. For any questions please contact info@victoriametrics.com
2023-12-21T08:28:44.253Z	error	VictoriaMetrics/lib/promscrape/config.go:466	skipping "/tmp/foo.yaml" at `scrape_config_files` because of failure to parse it: yaml: unmarshal errors:
  line 1: cannot unmarshal !!str `asdf` into []*promscrape.ScrapeConfig
2023-12-21T08:28:44.253Z	info	VictoriaMetrics/app/vmagent/main.go:121	all the configs are ok; exiting with 0 status code
```

### Version

vmagent-20231212-225303-tags-v1.96.0-enterprise-0-ga771bc255

**Comment by hagen1778** [comment]
Related to https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4959

**Comment by valyala** [comment]
@lauri-paypay , `vmagent` should return error when it encounters invalid [scrape_config_files](https://docs.victoriametrics.com/vmagent.html#loading-scrape-configs-from-multiple-files) starting from the commit 3ac44baebe9e232d02ce47899d61055787fb12e6 . The check for invalid `scrape_config_files` can be disabled by running `vmagent` with `-promscrape.config.strictParse=false` command-line flag.

This commit will be included in the next release. In the mean time you can build `vmagent` from this commit according to [these docs](https://docs.victoriametrics.com/vmagent.html#how-to-build-from-sources) and verify whether it works as intended.

**Comment by lauri-paypay** [comment]
@valyala thanks. I confirmed that the fixed behavior is what I expected.

**Comment by valyala** [comment]
@lauri-paypay , thanks for the confirmation!

The bugfix has been included in [v1.97.0 release](https://docs.victoriametrics.com/CHANGELOG.html). Closing the issue as fixed.
